### PR TITLE
Modules for Blobs and Queues Encryption

### PR DESCRIPTION
**Pull request by alzimmermsft**
Creates new modules for Blobs (azure-storage-blob-cryptography) and Queues (azure-storage-queue-cryptography) encryption and creates a new package in the Blobs modules for specialized blob clients (Append, Block, and Page blob).

### CODE REVIEW

File: eng/spotbugs-aggregate-report/pom.xml [hunk 1]
```diff
@@ -27,11 +27,14 @@
     <azure-identity.version>1.0.0-preview.3</azure-identity.version>
     <azure-keyvault.version>4.0.0-preview.3</azure-keyvault.version>
     <azure-messaging-eventhubs.version>5.0.0-preview.3</azure-messaging-eventhubs.version>
-    <azure-messaging-eventhubs-checkpointstore-blob.version>1.0.0-preview.1</azure-messaging-eventhubs-checkpointstore-blob.version>
+    <azure-messaging-eventhubs-checkpointstore-blob.version>1.0.0-preview.1
+    </azure-messaging-eventhubs-checkpointstore-blob.version>
```
Comment (JonathanGiles) [inline eng/spotbugs-aggregate-report/pom.xml]: Line wrap necessary?

File: pom.client.xml [hunk 2]
```diff
@@ -402,6 +404,10 @@
                 <title>Azure Storage - Blobs</title>
                 <packages>com.azure.storage.blob*</packages>
               </group>
+              <group>
+                <title>Azure Storage - Blobs Cryptography</title>
```
Comment (JonathanGiles) [inline pom.client.xml]: I'd personally put a hyphen after 'Blobs' (and same with the Queues line below).

File: sdk/storage/azure-storage-blob-cryptography/pom.xml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      <groupId>org.spockframework</groupId>
+      <artifactId>spock-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.projectreactor</groupId>
+      <artifactId>reactor-test</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
```
Comment (JonathanGiles) [inline sdk/storage/azure-storage-blob-cryptography/pom.xml]: Just bring in blob and get the rest transitively.

File: eng/spotbugs-aggregate-report/pom.xml (hunk 1 above)
Comment (alzimmermsft) [reply eng/spotbugs-aggregate-report/pom.xml]: Could be wrapping due to a 120 character limit on the auto format functionality IntelliJ offers, will bump that up for XMLs

File: pom.client.xml (hunk 2 above)
Comment (alzimmermsft) [reply pom.client.xml]: Done

File: eng/spotbugs-aggregate-report/pom.xml [hunk 4]
```diff
@@ -70,18 +72,20 @@
...
-<!--                <source>..\..\sdk\keyvault\azure-keyvault-secrets\src\main\java</source>-->
-<!--                <source>..\..\sdk\keyvault\azure-keyvault-secrets\src\samples\java</source>-->
+                <!--                <source>..\..\sdk\keyvault\azure-keyvault-keys\src\main\java</source>-->
+                <!--                <source>..\..\sdk\keyvault\azure-keyvault-keys\src\samples\java</source>-->
+                <!--                <source>..\..\sdk\keyvault\azure-keyvault-secrets\src\main\java</source>-->
+                <!--                <source>..\..\sdk\keyvault\azure-keyvault-secrets\src\samples\java</source>-->
                 <source>..\..\sdk\storage\azure-storage-common\src\main\java</source>
                 <source>..\..\sdk\storage\azure-storage-blob\src\main\java</source>
                 <source>..\..\sdk\storage\azure-storage-blob\src\samples\java</source>
+                <source>..\..\sdk\storage\azure-storage-blob-cryptography\src\main\java</source>
```
[the remark this answers is not in the document]
Comment (gapra-msft) [reply eng/spotbugs-aggregate-report/pom.xml]: I can go ahead and do this in my Encryption PR

**Pull request merged**